### DISCUSSION TITLE
minio bucket quota behavour

### DISCUSSION THREAD

**Discussion by szabokacsa** [minio/minio#14324] [Q&A] [answered]
Hi, 
I tried to set 35MB qouta on a bucket, but it does not block the file upload when the bucket reached 35MB size.
It blocked the upload when the bucket reached around 44MB size.

This is the showcase what I made:
![image](https://user-images.githubusercontent.com/3647735/154259786-adc09112-4020-476b-acc8-436cb62651b5.png)

I have an other case. The quota is set to 30MB, but the upload is not blocked even when a 35MB big file is uploaded and the upload is not blocked even when a 100MB big file is uploaded.
![image](https://user-images.githubusercontent.com/3647735/154260165-67e48d99-a655-4498-852b-3cec93ab7d9f.png)

![image](https://user-images.githubusercontent.com/3647735/154260232-f5d2a2a4-baac-4719-8504-7466be2c77c1.png)

My question is, how should the qouta work?
I worked based on this [page](https://docs.min.io/minio/baremetal/reference/minio-mc-admin/mc-admin-bucket-quota.html)

minio image version: minio/minio:RELEASE.2022-01-03T18-22-58Z

**Comment by harshavardhana** [comment, marked as answer]
Quota is the best effort based on scanner bucket usage calculation. Quota applies after this usage has been calculated.

**Reply by szabokacsa** [reply]
I don't really understand what you mean.
Based on the [description](https://docs.min.io/minio/baremetal/reference/minio-mc-admin/mc-admin-bucket-quota.html) of hard quota: Hard quotas prevent the bucket size from growing past the specified limit.
Why does it allow to upload files after the limit was reached?

**Reply by klauspost** [reply]
MinIO is distributed and does not have a central database it can just "ask" for the size of a bucket. There can be hundreds of millions of objects in a bucket, so getting an accurate size on every single upload isn't feasible.

If MinIO sees that the last usage scan plus your new object exceeds your quota, it will refuse the upload. Usage is updated continuously, but may lag significantly behind since it requires scanning your objects.
